src/timer_factory.rs

```diff
@@ -107,13 +111,76 @@ pub struct Timeouts {
     pub mailbox_response: Duration,
 }
 
+/// The kinds of timeouts that can be awaited for an EtherCAT bus.
+///
+/// See [`Timeouts`] for what each timeout is.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub enum TimeoutKind {
```

Comment:
This enum is also used to report errors to the user, however the `WaitLoopDelay` doesn't make sense as it can never become an error. Could we make a separate `Timeout` enum in `error.rs` and expose that instead, then keep `TimeoutKind` only `pub(crate)`? This would also mean some doc comments can be added to all the variants in that error enum that discuss the failure that caused the timeout. For example,

```rust
/// Failed to transition state, e.g. SAFE-OP to OP.
```

Doesn't make sense on a `TimeoutKind` variant when there's not an error.

If that makes sense?

Reply:
I've elected to link back to `Timeouts` instead prognosticating the timeouts at the error so there's only one place where the nature of the timeouts are described.